zproject/backends.py

```diff
@@ -1500,6 +1514,17 @@ def auth_complete(self, *args: Any, **kwargs: Any) -> Optional[HttpResponse]:
 
         return result
 
+    @classmethod
+    def validate_idp_for_subdomain(cls, idp_name: str, subdomain: str) -> bool:
+        idp_dict = settings.SOCIAL_AUTH_SAML_ENABLED_IDPS.get(idp_name)
+        if idp_dict is None:  # nocoverage
+            # This is defensive code and should never happen.
+            return False
```

Comment:
If it's supposed to be impossible from the code structure, we should just `raise AssertionError()` rather than doing a `# nocoverage` comment.  Or we could do a simple test that just calls `validate_idp_for_subdomain('invalid', 'realm'), but the former might be better.

Reply:
I guess this could also be the caller's job; in one call point, at least, we already have the `idp_dict`.

Reply:
Changed in the original commit